### PR TITLE
valkey-benchmark: fix -r option to support keyspace values above INT_MAX

### LINKED ISSUE #4228: [BUG] valkey-benchmark: -r option silently fails for values above INT_MAX
**Describe the bug**
The -r (keyspacelen) parameter in valkey-benchmark is stored as an int and parsed with atoi. If a value larger than 2,147,483,647 is used, it is silently truncated to 32 bits. This produces an incorrect keyspace (and, for values like 2³², collapses to 0 → a single key). No error is reported. The key format reserves 12 digits (key:012345678901), but the current implementation can only ever use values up to 2,147,483,647 — a fraction of the range the format supports.

**To reproduce**
`valkey-benchmark -t set -r 4294967296 -n 100`
Expected: benchmark runs with the requested 4,294,967,296-key keyspace (or rejects with a clear error).
Actual: keyspacelen is a 32-bit int, so the value truncates to 0. keyspacelen silently becomes 0 and the benchmark uses a single key, with no warning. This is observable on the server: after the run, `DBSIZE` returns 1 and `KEYS *` shows the single key `key:000000000000` (all 100 SETs collapsed onto key 0), instead of the ~100 distinct keys a 4-billion-key keyspace should produce.

**Expected behavior**
Accept values up to 999,999,999,999, as the key format (key:012345678901) has 12 digits. Values beyond that limit, or non-numeric input, should produce a clear error message.

**Additional information**
- The 12-digit key format (`__rand_int__`) supports up to 999,999,999,999 but the `int` type caps at 2,147,483,647
- Key generation uses `random()` which is bounded at 2,147,483,647, limiting actual key range even with a change to `long` type

### PR DESCRIPTION
**Pull request by michellee-10**
Fixes #4228

## Changes
  - Convert `config.keyspacelen` from `int` to `long long` (64-bit on all platforms)
  - Replace `atoi` with `strtoll` for proper overflow and input validation
  - Widen RNG from `random()` (31-bit) to 62-bit by combining two calls
  - Cap accepted values at 999,999,999,999 (12-digit key limit)
  - Propagate `long long` through dataset and fuzzer function signatures
  - Add integration tests for overflow rejection and large keyspace key generation

## Design note
`random()` only gives 31 bits, so two calls are combined for 62-bit range. A 64-bit random number generator (`mt19937-64.c`) is already included in valkey-benchmark, but its state is shared and unlocked — using it from `--threads` or the fuzzer's per-client threads would race. Stuck with two `random()` calls instead since it's already thread-safe.

### CODE REVIEW

File: src/valkey-benchmark.c [hunk 1]
```diff
@@ -1793,14 +1793,20 @@ int parseOptions(int argc, char **argv) {
...
-                if (*p < '0' || *p > '9') goto invalid;
+            const char *next = argv[++i];
+            char *endptr;
+            errno = 0;
+            long long val = strtoll(next, &endptr, 10);
+            if (endptr == next || *endptr != '\0') {
+                fprintf(stderr, "Invalid value for -r: '%s' is not a valid number\n", next);
+                exit(1);
+            }
+            if (errno == ERANGE || val < 0 || val > 999999999999LL) {
```
Comment (sarthakaggarwal97) [inline src/valkey-benchmark.c]: can we define this as constant

File: src/valkey-benchmark.c [hunk 2]
```diff
@@ -1793,14 +1793,20 @@ int parseOptions(int argc, char **argv) {
...
-            const char *next = argv[++i], *p = next;
-            if (*p == '-') {
-                p++;
-                if (*p < '0' || *p > '9') goto invalid;
+            const char *next = argv[++i];
+            char *endptr;
+            errno = 0;
+            long long val = strtoll(next, &endptr, 10);
+            if (endptr == next || *endptr != '\0') {
+                fprintf(stderr, "Invalid value for -r: '%s' is not a valid number\n", next);
```
Comment (sarthakaggarwal97) [inline src/valkey-benchmark.c:1804]: can we add a test for this case as well?

File: src/fuzzer_command_generator.c [hunk 3]
```diff
@@ -1190,14 +1190,14 @@ static void addKeysToCommand(FuzzerCommand *cmd, int numkeys, CommandArgument *a
     }
 
     for (int i = 0; i < numkeys; i++) {
-        int keyNumber = rand() % fuzz_ctx->max_keys;
+        long long keyNumber = (long long)(((uint64_t)random() << 31) | (uint64_t)random()) % fuzz_ctx->max_keys;
```
Comment (sarthakaggarwal97) [inline src/fuzzer_command_generator.c]: `((uint64_t)random() << 31) | (uint64_t)random()` is used in four places. Maybe worth putting as a helper function in a common file. Maybe `cli_common` or something better? This is not a blocker though.

**Review by sarthakaggarwal97 — COMMENTED** [review]
Thanks for the change. The code looks mostly good, some minor comments.

**Review by sarthakaggarwal97 — APPROVED** [review]
